File: demo/android/app/src/main/kotlin/test/demo/sdkv2/ios/ia_sdk/IaClientMethods.kt
package test.demo.sdkv2.ios.ia_sdk

import android.content.Intent
import de.ihreapotheken.sdk.integrations.api.IaSdk
import de.ihreapotheken.sdk.ordering.OrderingModule
import de.ihreapotheken.sdk.otc.OtcModule
import de.ihreapotheken.sdk.pharmacy.PharmacyModule
import de.ihreapotheken.sdk.rx.RxModule
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel

/**
 * Flutter client service call handler.
 */
internal class IaClientMethods(
    private val bindings: IaClientBindings,
) {
    /**
     * Collection of available method invocation identifiers.
     */
    @Suppress("EnumEntryName")
    enum class FlutterCall {
        /**
         * Allocates the SDK runtime resources.
         */
        initIaSdk,

        /**
         *
         */
        startComposeActivity,
    }

    fun callHandler(
        call: MethodCall,
        result: MethodChannel.Result,
    ) {
        when (call.method) {
            FlutterCall.initIaSdk.name -> {
                val args = call.arguments
                if (args !is Map<*, *>) {
                    result.error(
                        "ARG_ERROR",
                        "Arguments for initIaSdk must be of Dictionary type.",
                        null,
                    )
                    return
                }
                val accessKey = args["accessKey"] as? String
                val clientId = args["clientId"] as? String
                val serverEnvString = args["serverEnvironment"] as? String
                if (accessKey == null || clientId == null || serverEnvString == null) {
                    result.error(
                        "ARG_ERROR",
                        "Missing or invalid argument types. Expected String values for accessKey, clientId, and serverEnvironment.",
                        null,
                    )
                    return
                }
                val sdkModule = IaSdk.register(
                    OtcModule,
                    OrderingModule,
                    PharmacyModule,
                    RxModule,
                )
                sdkModule.init(
                    context = bindings.applicationContext,
                    apiKey = accessKey,
                    clientID = clientId,
                )
                result.success(null)
            }

            FlutterCall.startComposeActivity.name -> {
                val intent = Intent(bindings.applicationContext, IaClientComponentActivity::class.java)
                intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
                bindings.applicationContext.startActivity(intent)
                result.success(null)
            }

            else -> {
                result.notImplemented()
            }
        }
    }
}

File: demo/ios/Runner/IaSdk/IaClientMethods.swift
import IACore
import IAIntegrations
import IAOverTheCounter
import IAOrdering

/**
 * Flutter client service call handler.
 */
@MainActor
internal class IaClientMethods {
  /**
   * Collection of available method invocation identifiers.
   */
  enum FlutterCall : CaseIterable {
    /**
     * Allocates the SDK runtime resources.
     */
    case initIaSdk
    
    /**
     * String identifier getter definition.
     */
    var name: String {
      return String(describing: self)
    }
  }
  
  /**
   * Flutter SDK host app bindings definitions.
   */
  private let bindings: IaClientBindings!
  
  init(bindings: IaClientBindings!) {
    self.bindings = bindings
  }
  
  /**
   * Registers a handler for method calls from the Flutter side.
   */
  func callHandler(call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case FlutterCall.initIaSdk.name:
      let args = call.arguments
      guard 
        let args = args as? [String: Any] 
      else {
        return result(
          FlutterError(
            code: "ARG_ERROR",
            message: "Arguments for initIaSdk must be of Dictionary type.",
            details: nil
          )
        )
      }
      guard
        let accessKey = args["accessKey"] as? String,
        let clientId = args["clientId"] as? String,
        let serverEnvString = args["serverEnvironment"] as? String
      else {
        return result(
          FlutterError(
            code: "ARG_ERROR",
            message: "Missing or invalid argument types. Expected String values for accessKey, clientId, and serverEnvironment.",
            details: nil
          )
        )
      }
      guard 
        let serverEnvironment = EnvironmentID(rawValue: serverEnvString) 
      else {
        return result(
          FlutterError(
            code: "ARG_ERROR",
            message: "Invalid environment ID: \(serverEnvString)",
            details: nil
          )
        )
      }
      IASDK.configuration.apiKey = accessKey
      IASDK.configuration.clientID = clientId
      IASDK.setEnvironment(serverEnvironment)
      IASDK.delegate = bindings.delegate
      IAIntegrationsSDK.register()
      IAOverTheCounterSDK.register()
      IAOrderingSDK.register(delegate: bindings.delegate)
      Task.init {
        do {
            let _ = try await IASDK.initialize(options: .init(
              prerequisitesOptions: IASDKPrerequisitesOptions(
                shouldShowIndicator: true,
                isCancellable: false,
                isAnimated: true)
              ),
            )
            result(nil)
        } catch {
          result(
            FlutterError(
              code: "INIT_ERROR",
              message: error.localizedDescription,
              details: nil)
          )
        }
      }
      break
    default:
      return result(FlutterMethodNotImplemented)
    }
  }
}
